### PR TITLE
dnsdist: Add a Lua maintenance hook

### PR DESCRIPTION
**Pull request by rgacogne**
### Short description

Having a hook that can accept multiple callbacks makes it a lot easier to manage the execution of several, independant tasks that needs to be called regularly.

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [x] tested this code
- [x] included documentation (including possible behaviour changes)
- [ ] documented the code
- [x] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

File: pdns/dnsdistdist/dnsdist-lua-hooks.cc [hunk 1]
```diff
@@ -0,0 +1,32 @@
+
+#include "dnsdist-lua-hooks.hh"
+#include "dnsdist-lua.hh"
+#include "lock.hh"
+
+namespace dnsdist::lua::hooks
+{
+static LockGuarded<std::vector<MaintenanceCallback>> s_maintenanceHook;
+
+void runMaintenanceHook(const LuaContext& context)
```
Comment (chbruyand) [inline pdns/dnsdistdist/dnsdist-lua-hooks.cc:10]:
```suggestion
void runMaintenanceHooks(const LuaContext& context)
```
Maybe ?

**Review by chbruyand — APPROVED** [review]
Just a small nit that is a matter of taste, LGTM otherwise

File: pdns/dnsdistdist/dnsdist-lua-hooks.cc (hunk 1 above)
Comment (rgacogne) [reply pdns/dnsdistdist/dnsdist-lua-hooks.cc]: Done!

**Comment by rgacogne** [comment]
The recursor regression failure can be ignored (the test uses an external DNS zone which is currently broken).
As for `clang-tidy` warning about a potential leak in LuaWrapper's code, which was not modified by this pull request, it seems like a false positive to me: 
```
ext/luawrapper/include/LuaContext.hpp:2888 Potential memory leak (clang-analyzer-cplusplus.NewDeleteLeaks)
```

**Review by chbruyand — APPROVED** [review]

**Pull request merged**